src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/Ordering/AbstractMoveCommand.cs

```diff
@@ -41,76 +41,40 @@ public AbstractMoveCommand(IPhysicalProjectTree projectTree, SVsServiceProvider
             return await GetCommandStatusResult.Handled(commandText, CommandStatus.Enabled).ConfigureAwait(true);
         }
 
-        protected override async Task<bool> TryHandleCommandAsync(IProjectTree node, bool focused, long commandExecuteOptions, IntPtr variantArgIn, IntPtr variantArgOut)
+        private async Task<TResult> OpenProjectForWriteAsync<TResult>(ConfiguredProject project, Func<Project, TResult> action)
         {
-            var didMove = await TryMoveAsync(_configuredProject, node).ConfigureAwait(true);
+            Requires.NotNull(project, nameof(project));
+            Requires.NotNull(project, nameof(action));
 
-            if (didMove)
-            {
-                // Wait for updating to finish before re-selecting the node that moved.
-                // We need to re-select the node after it is moved in order to continuously move the node using hotkeys.
-                await _projectTree.TreeService.PublishLatestTreeAsync(waitForFileSystemUpdates: true).ConfigureAwait(true);
+            var projectLockService = _configuredProject.Services.ProjectService.Services.ProjectLockService;
 
-                if (_configuredProject.UnconfiguredProject.Services.HostObject is IVsUIHierarchy hierarchy)
-                {
-                    // The node moved will retain its identity.
-                    var itemId = (uint)node.Identity.ToInt32();
-                    var window = GetUIHierarchyWindow(_serviceProvider, Guid.Parse(ManagedProjectSystemPackage.SolutionExplorerGuid));
+            using (ProjectWriteLockReleaser access = await projectLockService.WriteLockAsync())
+            {
+                await access.CheckoutAsync(project.UnconfiguredProject.FullPath)
+                            .ConfigureAwait(true);
 
-                    window.ExpandItem(hierarchy, itemId, EXPANDFLAGS.EXPF_SelectItem);
+                Project evaluatedProject = await access.GetProjectAsync(project)
+                                                       .ConfigureAwait(true);
 
-                    // If we moved a folder, collapse it.
-                    if (node.IsFolder)
-                    {
-                        window.ExpandItem(hierarchy, itemId, EXPANDFLAGS.EXPF_CollapseFolder);
-                    }
-                }
+                // Deliberately not async to reduce the type of
```

Comment:
this doesn't prevent async code. Consumers can always `JTF.Run` to execute async code within this. Which is even worse because that is a synchronous block. Your only option here is documentation to warn callers to not do too much within a lock.

Reply:
This is a terrible comment and this whole thing has been addressed in another PR.